src/main/java/core/basesyntax/strategy/ReportGenerateImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class ReportGenerateImpl implements ReportGenerate {
```

Comment:
The class name 'ReportGenerateImpl' could be more descriptive. Consider renaming it to reflect that it generates a report based on the current state of the Storage, for example, 'StorageReportGenerator'.